### PR TITLE
[Snackbar] Changed the background to use a MaterialShapeDrawable to allow shape customizations

### PR DESCRIPTION
**Pull request by gabrielemariotti**
According to material design [documentation](https://material.io/design/shape/applying-shape-to-ui.html#shape-scheme) the `Snackbar` can support customized shapes.

This PR changes the current background to use a `MaterialShapeDrawable`.
To avoid breaking changes it supports also the `android:background` attribute with a logic similar to the `MaterialButton` (for example with the `Widget.Design.Snackbar` style).

You can use the demo catalog to test the different shapes:

Default:
Schermata 2020-09-11 alle 19 35 27

Rounded - Large:
Schermata 2020-09-11 alle 19 07 36

Cut - Medium:
Schermata 2020-09-11 alle 19 07 44

Custom rounded - 50%
Schermata 2020-09-11 alle 19 11 39

### CODE REVIEW

**Comment by raajkumars** [comment]
@gabrielemariotti Thanks for the PR. I would really appreciate it you can please resolve the conflicts in styles.xml

**Comment by gabrielemariotti** [comment]
@raajkumars Thanks for your feedback. Conflict resolved.

File: lib/java/com/google/android/material/snackbar/BaseTransientBottomBar.java [hunk 1]
```diff
@@ -1127,69 +1134,136 @@ protected SnackbarBaseLayout(@NonNull Context context, AttributeSet attrs) {
...
+      super.setBackgroundDrawable(background);
+    }
+
     @Override
     public void setBackgroundDrawable(@Nullable Drawable drawable) {
-      if (drawable != null && backgroundTint != null) {
-        drawable = DrawableCompat.wrap(drawable.mutate());
-        DrawableCompat.setTintList(drawable, backgroundTint);
-        DrawableCompat.setTintMode(drawable, backgroundTintMode);
+      if (isUsingOriginalBackground()){
```
Comment (drchen) [inline lib/java/com/google/android/material/snackbar/BaseTransientBottomBar.java:1204]: Don't really think we should have the if-else clauses here. 

We will always need L1207~1212 to be executed I think.

File: lib/java/com/google/android/material/snackbar/BaseTransientBottomBar.java [hunk 2]
```diff
@@ -1127,69 +1134,136 @@ protected SnackbarBaseLayout(@NonNull Context context, AttributeSet attrs) {
...
-        DrawableCompat.setTintList(wrappedBackground, backgroundTint);
-        DrawableCompat.setTintMode(wrappedBackground, backgroundTintMode);
-        if (wrappedBackground != getBackground()) {
-          super.setBackgroundDrawable(wrappedBackground);
+      if (isUsingOriginalBackground()) {
+        if (materialBackgroundDrawable != null) {
+          DrawableCompat.setTintList(materialBackgroundDrawable, backgroundTint);
+        }
+      } else {
+        if (getBackground() != null) {
```
Comment (drchen) [inline lib/java/com/google/android/material/snackbar/BaseTransientBottomBar.java:1226]: Can be combined into else if {

File: lib/java/com/google/android/material/snackbar/BaseTransientBottomBar.java [hunk 3]
```diff
@@ -1127,69 +1134,136 @@ protected SnackbarBaseLayout(@NonNull Context context, AttributeSet attrs) {
...
-      if (getBackground() != null) {
-        Drawable wrappedBackground = DrawableCompat.wrap(getBackground().mutate());
-        DrawableCompat.setTintMode(wrappedBackground, backgroundTintMode);
-        if (wrappedBackground != getBackground()) {
-          super.setBackgroundDrawable(wrappedBackground);
+      if (isUsingOriginalBackground()) {
+        if (materialBackgroundDrawable != null) {
+          DrawableCompat.setTintMode(materialBackgroundDrawable, backgroundTintMode);
+        }
+      } else {
```
Comment (drchen) [inline lib/java/com/google/android/material/snackbar/BaseTransientBottomBar.java:1244]: ditto

**Comment by dsn5ft** [comment]
Thanks for the PR! This was addressed in https://github.com/material-components/material-components-android/commit/a71781355e48eb6106fc42bd4237027b71e8d803.

**Pull request closed without merging**